### PR TITLE
Hashmap repeated word

### PR DESCRIPTION
**Pull request by JamesBrooks01**

### CODE REVIEW

**Comment by JamesBrooks01** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
       - [x] Summary, Description, Approach & Efficiency, Solution
       - [x] Picture of whiteboard
       - [x] Link to code
 - [x] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
       - [x] “Happy Path” - Expected outcome
       - [x] Expected failure
       - [ ] Edge Case (if applicable/obvious)

File: python/docs/hashtable_repeated_word/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Challenge Summary
+
+Our task is to write out a function that takes in a string and returns the word that repeats first within it. I'm pretty sure I'm pushing the limits of what could be acceptable for the solution to this problem, but it does work.
```
Comment (chloenott) [inline python/docs/hashtable_repeated_word/README.md:3]: It's a super interesting solution!

File: python/code_challenges/hashtable_repeated_word.py [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+    split = punct_removed.split(' ')
+    counted = Counter(split)
+    first_repeat = None
+    num_to_zero_two = 100
+    for x in counted:
+        if counted[x] > 1:
+            first_index = split.index(x)
+            split.pop(first_index)
+            second_index = split.index(x)
+            split.insert(first_index, x)
```
Comment (chloenott) [inline python/code_challenges/hashtable_repeated_word.py:18]: I think `list.insert()` is O(n). Something to think about because this is inside a for loop.

Also, interestingly, `.pop` at a specified index is not O(1). Which is a little counterintuitive because pop without anything is O(1).

File: python/code_challenges/hashtable_repeated_word.py [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+from collections import Counter
+import re
 
+def first_repeated_words(str):
+    if str == '':
+        return None
+    lowercase_input = str.lower()
+    punct_removed = re.sub('[^\w\s\’]','',lowercase_input)
+    split = punct_removed.split(' ')
+    counted = Counter(split)
```
Comment (chloenott) [inline python/code_challenges/hashtable_repeated_word.py:10]: A counter! Cool!

File: python/code_challenges/hashtable_repeated_word.py [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+    first_repeat = None
+    num_to_zero_two = 100
```
Comment (chloenott) [inline python/code_challenges/hashtable_repeated_word.py:12]: This variable is really tricky to think of a readable name for! I think `num_to_zero_two` is a decent compromise given the circumstance. Maybe `current_best_second_index`?

**Review by chloenott — COMMENTED** [review]
Super neat solution! Comments below.
